specreduce/tracing.py

```diff
@@ -177,24 +175,31 @@ class KosmosTrace(Trace):
         guess position. Useful for tracing faint sources if multiple
         traces are present, but potentially bad if the trace is
         substantially bent or warped. [default: None]
+    trace_model : `~astropy.modeling.Model` or, optional
+        The 1-D polynomial model used to fit the trace to the bins' peak
+        pixels. Valid options are ``Spline1D``, ``Legendre1D``,
+        ``Chebyshev1D``, and ``Polynomial1D``. ``Spline1D`` is used with
```

Comment:
this makes it sound like the strings are accepted.  Can we use intersphinx links to link to the astropy classes themselves?

But maybe it would also be useful to support passing the string names in which case the default options for each model would be used?  Or we could leave that up to either classmethods or subclasses:

```
AutoTrace.polynomial(image, order=2, ...)
```

or

```
PolyTrace(image, order=2, ...) # which is just a subclass of AutoTrace except that it accepts input arguments specfic to that model
```

Reply:
I changed their representation in the docstring. It looks like consensus moved away from the subclass approach.

Passing in strings could be helpful, but I would put it below a few other issues for the time being... I'll keep this unresolved and reassess closer to merge.